components/ide/jetbrains/backend-plugin/src/main/kotlin/io/gitpod/jetbrains/remote/GitpodTerminalService.kt

```diff
@@ -164,9 +174,45 @@ class GitpodTerminalService(session: ClientProjectSession) {
     }
 
     private fun createAttachedSharedTerminal(supervisorTerminal: TerminalOuterClass.Terminal) {
-        createSharedTerminalAndExecuteCommand(
+        val shellTerminalWidget = createSharedTerminalAndExecuteCommand(
             supervisorTerminal.title,
             "gp tasks attach ${supervisorTerminal.alias}"
+        ) ?: return
+
+        shellTerminalWidget.addListener {
+            terminalServiceFutureStub.shutdown(
+                    TerminalOuterClass.ShutdownTerminalRequest.newBuilder()
+                            .setAlias(supervisorTerminal.alias)
+                            .build()
+            )
+        }
```

Comment:
ℹ️ This code:
```kotlin
        shellTerminalWidget.addListener {
        
        }
```
is a simplification (suggested by IntelliJ itself) of this other code:
```kotlin
        shellTerminalWidget.addListener(object: TerminalWidgetListener {
            override fun allSessionsClosed(widget: TerminalWidget) {

            }
        })
```

Reply:
I think allSessionsClosed is more cleaner

Reply:
Ok, I've left the cleaner version.
I had to add a `@Suppress("ObjectLiteralToLambda")` statement, otherwise, the IDE would keep warning about it.